src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  isAlive() {
+    return this.health > 0;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (victim instanceof Herbivore && !victim.hidden) {
+      victim.health -= 50;
+    }
+
+    if (victim.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== victim);
+    }
```

Comment:
The condition `if (victim.health <= 0)` checks if the victim's health is less than or equal to zero to remove it from the `Animal.alive` array. However, if the victim's health is exactly zero after a bite, it will still be removed. Ensure this behavior is intended, as it might be more appropriate to check for `victim.health < 0` to only remove the victim when its health is negative.